### ISSUE TITLE
OGM batch performance problems due to string concatenation

### ISSUE THREAD

**Issue by k-sok** [mogui/pyorient#261]
The resulting batch script is built by using string concatenation.

```
commands = 'BEGIN\n'
commands += cmd1
commands += cmd2
...
commands += cmdN
g.client.batch(commands)
```
The larger the entire script is (i.e. the commands string) the longer takes each concatenation operation (because of the underlying memory reallocation and copying). 

A better approach is to use a list of strings and join it when committing.

```
commands = ['BEGIN']
commands.append(cmd1)
commands.append(cmd2)
...
commands.append(cmdN)
g.client.batch('\n'.join(commands)
```

By implementing the above solution, I was able to improve the performance with large batches by magnitudes.

**Comment by TropicalPenguin** [comment]
Yo,

The develop branch already uses the list strategy :)

**Comment by k-sok** [comment]
Yes, just saw it after forking. I wonder how are the chances that this stack design will survive to the next version. I have to subclass `Batch` and have to decide what to rely on...
